src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -353,6 +358,17 @@ winrt::hstring Profile::EvaluatedStartingDirectory() const
     return path;
 }
 
+void Profile::_FinalizeInheritance()
+{
+    if (auto defaultAppearanceImpl = get_self<AppearanceConfig>(_DefaultAppearance))
+    {
+        if (auto parentDefaultAppearanceImpl = _parents.at(0)->_DefaultAppearance.try_as<AppearanceConfig>())
+        {
+            defaultAppearanceImpl->InsertParent(0, parentDefaultAppearanceImpl);
+        }
+    }

```

Comment:
Do we need to do this for every parent? Some profiles have two parents. This might result in a generated profile having some of its appearance settings ignored.

Reply:
Thanks for the catch! Fixed